modules/nextflow/src/main/groovy/nextflow/executor/BashFunLib.groovy

```diff
@@ -95,7 +95,11 @@ class BashFunLib<V extends BashFunLib> {
             while ((i<\${#cmd[@]})); do
                 local copy=()
                 for x in "\${pid[@]}"; do
-                  [[ -e /proc/\$x ]] && copy+=(\$x)
+                  if [[ -e /proc/\$x ]]; then
+                    copy+=(\$x)   # process still exists, remember it
+                  else
+                    wait \$x      # process exited, wait on it
+                  fi
```

Comment:
```suggestion
                  # if the process exist, keep in the 'copy' array, otherwise wait on it to capture the exit code
                  # see https://github.com/nextflow-io/nextflow/pull/4050 
                  [[ -e /proc/\$x ]] && copy+=(\$x) || wait \$x 
```

Reply:
I'd suggest a more compact notation